### ISSUE TITLE
New rule request: Shadows name from outer scope 

### ISSUE THREAD

**Issue by Garrett-R** [astral-sh/ruff#6903]
What about catching this code smell:
```python
def func(x: list[int]) -> list[int]:
    return sorted(x, key=lambda x: x)
```
(PyCharm warns "Shadows name 'x' from outer scope")

BTW, we're loving Ruff!  Amazing project.

**Comment by DetachHead** [comment]
pylint has a rule for that (#970)

**Comment by Garrett-R** [comment]
Which rule exactly is that in pylint?  I can't seem to find one.

**Comment by DetachHead** [comment]
oh sorry, it's [`redefined-outer-name`](https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/redefined-outer-name.html)

**Comment by dylwil3** [comment]
Closing this as duplicate since it's included in both of these issues: #970 and #3040 

Thanks!

**Comment by ddorian** [comment]
Now we can't subscribe to an issue to know when it's "fixed" since those issues are wrappers for multiple rules.
